Funcky.Test/Monads/OptionEqualityComparerTest.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using Funcky.Monads;
+using Xunit;
+
+namespace Funcky.Test.Monads
+{
+    public class OptionEqualityComparerTest
+    {
+        [Fact]
+        public void TwoNoneOptionsAreEqual()
+        {
+            Assert.Equal(Option<int>.None(), Option<int>.None(), OptionEqualityComparer.Create(new ThrowingEqualityComparer<int>()));
+        }
+
+        [Theory]
+        [MemberData(nameof(SomeAndNoneAreNotEqualData))]
+        public void SomeAndNoneAreNotEqual(Option<int> x, Option<int> y)
+        {
+            Assert.NotEqual(x, y, OptionEqualityComparer.Create(new ThrowingEqualityComparer<int>()));
+        }
+
+        public static TheoryData<Option<int>, Option<int>> SomeAndNoneAreNotEqualData()
+            => new ()
+            {
+                { Option<int>.None(), Option.Some(13) },
+                { Option.Some(13), Option<int>.None() },
+                { Option<int>.None(), Option.Some(default(int)) },
+                { Option.Some(default(int)), Option<int>.None() },
+            };
+
+        [Fact]
+        public void GetHashCodePropagatesToTheItem()
+        {
+            Assert.Equal("foo".GetHashCode(), Option.Some("foo").GetHashCode());
+        }
+
+        [Theory]
+        [InlineData(10, 10)]
```

Comment:
Maybe a Property Test here? :) should be very simple to implement

Reply:
1e409a4a4f02da0c9bf13606f7ca77dab4d943d5